unit-tests/live/syncer/test-throughput.cpp

```diff
@@ -0,0 +1,248 @@
+            };
+
+            float delta = (float)std::chrono::duration_cast<std::chrono::seconds>(t_end - t_start).count();
+            while (delta < RECEIVE_FRAMES_TIME)
+            {
+                if (delta > RECEIVE_FRAMES_TIME/2)
+                    set_ir_exposure(); // set exposure to 18000 after 5 sec
+
+                auto fs = _sync.wait_for_frames();
+                for (const rs2::frame& ff : fs)
+                {
+                    process_frame(ff);
+                }
+                t_end = std::chrono::system_clock::now();
+                delta = (float)std::chrono::duration_cast<std::chrono::seconds>(t_end - t_start).count();
+            }
+        }  
+        void analysis(configuration test, std::vector <std::pair<float, float>>& frames_arrival_info, std::string stream_type = "Infrared 1")
+        {
+            if (frames_arrival_info.empty())
+                return;
+            float fps = 0.0f;
+            while (fps < 10) // skip unstable frames at the beginning
+            {
+                fps = frames_arrival_info.front().second;
+                frames_arrival_info.erase(frames_arrival_info.begin());
+            }
+            std::vector<std::pair<float, float>>  vec[2];
+            for (auto& frm : frames_arrival_info)
+            { 
+                auto fps_ratio = frm.second / fps; // used ratio instead of precised fps because syncer doesn't change fps to exact value when running RGB
+                if (fps_ratio > 0.9)
+                    vec[0].push_back(frm);
+                else
+                    vec[1].push_back(frm);
+            }
+
+            float delta[2] = { _delta, RECEIVE_FRAMES_TIME - _delta };
+            int i = 0;
+            for (auto& v : vec)
+            {
+                if (v.empty())
+                    continue;
+                auto actual_fps = v.front().second;
+                float calc_fps = (float)v.size() / delta[i];
+                float fps_ratio = calc_fps / actual_fps;
+                CAPTURE(stream_type, actual_fps, calc_fps, v.size());
+                CHECK(fps_ratio > 0.9);
+                check_frame_drops(v);
+                i += 1;
+            }
+        }
+        void check_frame_drops(std::vector<std::pair<float, float>> frames)
+        {
+            auto actual_fps = frames.front().second;
+            float prev_frame_time = frames.front().first;
+            frames.erase(frames.begin());
+            float expected_dt_ms = 1000/actual_fps; // convert sec -> msec
+            
+            for (auto& frame : frames)
+            {
+                auto frame_time = frame.first;
+                float calc_dt_msec = ((float)frame_time - prev_frame_time)/1000; //convert usec -> msec
+                float dt_msec_ratio = calc_dt_msec / expected_dt_ms;
+                CAPTURE(expected_dt_ms, calc_dt_msec, dt_msec_ratio, frame_time, prev_frame_time);
+                CHECK(dt_msec_ratio > 0.9);
```

Comment:
The test condition is reverted - anything bigger than 1.5 is definitely a frame drop.